examples/hex_neighbors.rs

```diff
@@ -96,23 +99,23 @@ fn spawn_tile_labels(
         color: Color::BLACK,
     };
     let text_alignment = TextAlignment::CENTER;
-    for (map_transform, map_type, grid_size, tilemap_storage) in tilemap_q.iter() {
+    for (map_type, grid_size, tilemap_storage) in tilemap_q.iter() {
         for tile_entity in tilemap_storage.iter().flatten() {
             let tile_pos = tile_q.get(*tile_entity).unwrap();
             let tile_center = tile_pos.center_in_world(grid_size, map_type).extend(1.0);
-            let transform = *map_transform * Transform::from_translation(tile_center);
-            commands
-                .entity(*tile_entity)
-                .insert(Text2dBundle {
+            let transform = Transform::from_translation(tile_center);
```

Comment:
Does this work because the tilemap is being centered now? 

Regardless, I am not sure if map_transform should be removed, because it is worth knowing this should be done in an example.

For example, what happens if a map is panned? Does the example still work?

Reply:
Fascinating question. The examples work just as they did before when the map is panned and when the map type is changed.

Although now that I am looking at this again, it does seem like the map transform *should* be involved here. (Despite `TilePos::center_in_world`'s claims about returning a world position, I don't think it is?)

I'll take a closer look at this in a bit.

Reply:
Okay, I did some investigating.

The `Text2dBundle`'s components (including GlobalTransform) are being added directly to the Tile, and the Tiles are all children of the Tilemap.

So the text's transform is relative to the Tilemap because transform propagation is happening. This all worked out fine when the Tilemap was at 0,0 prior to this PR. But now that the Tilemap is centered, this double-adding the map transform produces the wrong result.

I'm not sure what the "bevy_ecs_tilemap" way to fix this is. To me, it looks like the text should be spawned as a separate entity, and a component like `TileTextEntity(Entity)` should be inserted onto the Tile instead. It's my understanding that "bevy_ecs_tilemap" is generally avoiding any transform or visibility propagation on purpose for performance reasos.

If that solution sounds acceptable, I'd be happy to implement it.

Reply:
This is what that changeset would look like for one of the examples: <https://github.com/rparrett/bevy_ecs_tilemap/commit/1b83aeee0f6ea10295a395ad91febd85ff492242>

Reply:
Thanks for looking into this. 

IIUC, I think the correct answer here then is to remove `map_transform` being multiplied here, and trust that the text entity will be properly managed by bevy. 

Why? Because text is not something deeply a part of bevy_ecs_tilemap; it's just a debugging tool for the example. Likely, if players need more complex text solutions, they will have to roll their own. For now, for the sake of the example, then, there's no need to multiply by the map transform?

@StarArawn Let me know what you think.

Reply:
@rparrett Oh, I just had a look at the example you changed. So, does wrapping the `Text2dBundle` in its own component remove the transform propagation?

Reply:
I think you understand correctly (if I do.) The map_transform multiplication prior to this PR is a bug that wasn't revealed until the map was moved.

Adding the `Text2dBundle` outside of bevy's hierarchy and into "our own hierarchy" (storing a reference to the Entity on a component) prevents the transform propagation from happening if we want that behavior to work. (At the expense of complicating all the interactions with the text)

Displaying coordinates or other sprites or whatever on top of tiles is a common thing for users to want to do, so IMHO, we should use whatever pattern we want users to use.

I am an extremely casual user of `bevy_ecs_tilemap` that hasn't followed recent developments closely, so I don't have a very good intuition for what that is.

Reply:
@rparrett Okay, after talking about this with StarToaster, and thinking about it further myself, I think:

1) for now, to get the examples working, we can do what you suggested here: https://github.com/StarArawn/bevy_ecs_tilemap/pull/356#discussion_r1033070351

2) for future work, we should make an issue for TileText that follows your more general suggestion